pkg/apis/pubsub/v1alpha1/topic_defaults.go

```diff
@@ -0,0 +1,29 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import "context"
+
+func (t *Topic) SetDefaults(ctx context.Context) {
+	t.Spec.SetDefaults(ctx)
+}
+
+func (ts *TopicSpec) SetDefaults(ctx context.Context) {
```

Comment:
The comments in the OpenAPI schema above note that several fields are defaulted, but I don't see their defaults set here. Are they set elsewhere?

Reply:
Why set them in the controller? Is the default impossible/difficult to set in the webhook?